openghg/retrieve/icos/_retrieve.py

```diff
@@ -0,0 +1,305 @@
+# I'm creating this submodule as I'm not quite sure where else to put this for now
+# we can always move it in the future
+from pandas import DataFrame
+from typing import Dict, List, Optional, Union
+
+from openghg.dataobjects import ObsData
+
+
+def retrieve(
+    site: str,
+    species: Optional[Union[str, List]] = None,
+    sampling_height: Optional[str] = None,
+    start_date: Optional[str] = None,
+    end_date: Optional[str] = None,
+    force_retrieval: bool = False,
+    data_level: int = 2,
+) -> Union[ObsData, List[ObsData], None]:
+    """Retrieve ICOS atmospheric measurement data. If data is found in the object store it is returned. Otherwise
+    data will be retrieved from the ICOS Carbon Portal. Data retrieval from the Carbon Portal may take a short time.
+    If only a single data source is found an ObsData object is returned, if multiple a list of ObsData objects
+    if returned, if nothing then None.
+
+    Args:
+        site: Site code
+        species: Species name
+        start_date: Start date
+        end_date: End date
+        force_retrieval: Force the retrieval of data from the ICOS Carbon Portal
+        data_level: ICOS data level, see https://icos-carbon-portal.github.io/pylib/modules/#stationdatalevelnone
```

Comment:
May be worth paraphrasing what's included at the end this link e.g. 1= Near Real Time Data, 2=Final Quality Checked Data.